### PR TITLE
Veelay Tsahvahn Sky Relto Page

### PR DESCRIPTION
**Pull request by Hazado**
Relto Page of the Veelay Tsahvahn sky

### CODE REVIEW

File: Scripts/SDL/Personal.sdl [hunk 1]
```diff
@@ -2385,4 +2385,136 @@ STATEDESC Personal
...
+    VAR BOOL    psnlCalendarStone06[1]  DEFAULT=0   # June      - Descent (TreasureBookShaft)
+    VAR BOOL    psnlCalendarStone07[1]  DEFAULT=0   # July      - ?
+    VAR BOOL    psnlCalendarStone08[1]  DEFAULT=0   # August    - ?
+    VAR BOOL    psnlCalendarStone09[1]  DEFAULT=0   # September - ?
+    VAR BOOL    psnlCalendarStone10[1]  DEFAULT=0   # October   - ?
+    VAR BOOL    psnlCalendarStone11[1]  DEFAULT=0   # November  - ?
+    VAR BOOL    psnlCalendarStone12[1]  DEFAULT=0   # December  - ?
+## Guild Alliance SDL
+    VAR INT     guildAlliance[1]    DEFAULT=0
 }
```
Comment (Hoikas) [inline Scripts/SDL/Personal.sdl:2520]: Newline at EOF

File: Scripts/SDL/Personal.sdl [hunk 2]
```diff
@@ -2385,4 +2385,136 @@ STATEDESC Personal
...
+    VAR BOOL    psnlLibraryDoorClosed[1]    DEFAULT=1
+    VAR BOOL    ShelfABoolOperated[1]    DEFAULT=0
+    VAR INT    ShelfAUserID[1]    DEFAULT=0
+    VAR BOOL    psnlBlindsClosed[1] DEFAULT=1
+    VAR BOOL    psnlClosetClosed[1] DEFAULT=1
+    VAR BOOL    CleftVisited[1]     DEFAULT=0
+    VAR BOOL    psnlCleftTotemOpen[1]     DEFAULT=0
+    VAR BOOL    psnlCleftSolved[1]     DEFAULT=0
+    VAR BOOL    psnlGotCityBook[1]     DEFAULT=0
+    VAR STRING32    psnlMusicBoxCurrentSong[1]    DEFAULT=""
```
Comment (Hoikas) [inline Scripts/SDL/Personal.sdl:2405]: Can we cleanup the whitespace to be consistent in this block?

File: Scripts/SDL/Personal.sdl [hunk 3]
```diff
@@ -2385,4 +2385,136 @@ STATEDESC Personal
...
+    VAR BYTE    YeeshaPage1[1]  DEFAULT=0 # sun and moon
+    VAR BYTE    YeeshaPage2[1]  DEFAULT=0 # waterfall
+    VAR BYTE    YeeshaPage3[1]  DEFAULT=0 # hut decal / interior rug
+    VAR BYTE    YeeshaPage4[1]  DEFAULT=0 # hut roof (swap)
+    VAR BYTE    YeeshaPage5[1]  DEFAULT=0 # jumping pinnacles
+    VAR BYTE    YeeshaPage6[1]  DEFAULT=0 # dock
+    VAR BYTE    YeeshaPage7[1]  DEFAULT=0 # kickable physicals
+    VAR BYTE    YeeshaPage8[1]  DEFAULT=0 # Rain
+    VAR BYTE    YeeshaPage9[1]  DEFAULT=0 # music player
+    VAR BYTE    YeeshaPage10[1]  DEFAULT=0 # the tree
```
Comment (Hoikas) [inline Scripts/SDL/Personal.sdl:2418]: Whitespace

File: Scripts/SDL/Personal.sdl [hunk 4]
```diff
@@ -2385,4 +2385,136 @@ STATEDESC Personal
...
+
+   VAR BYTE  TeledahnPoleState[1]    DEFAULT=0
+   VAR BYTE  KadishPoleState[1]      DEFAULT=0
+   VAR BYTE  GardenPoleState[1]      DEFAULT=0
+   VAR BYTE  GarrisonPoleState[1]    DEFAULT=0
+
+   VAR BOOL  TeledahnEnabled[1]     DEFAULT=1 DEFAULTOPTION=VAULT
+   VAR BOOL  GarrisonEnabled[1]     DEFAULT=1 DEFAULTOPTION=VAULT
+   VAR BOOL  GardenEnabled[1]     DEFAULT=1 DEFAULTOPTION=VAULT
+   VAR BOOL  KadishEnabled[1]     DEFAULT=1 DEFAULTOPTION=VAULT
```
Comment (Hoikas) [inline Scripts/SDL/Personal.sdl:2469]: Whitespace

File: Scripts/SDL/Personal.sdl [hunk 5]
```diff
@@ -2385,4 +2385,136 @@ STATEDESC Personal
...
+
+# Other Stuff
+   VAR BOOL  psnlBugsVis[1]     DEFAULT=0 DEFAULTOPTION=VAULT
+
+# GlobalRewardClothing is a list of names of reward clothing to add to the player's closet, separated by semicolins
+# (do NOT use this for clothing items to be given to a select group of people)
+   VAR STRING32  GlobalRewardClothing[1]  DEFAULTOPTION = VAULT
+
+# Enables the GreatZero GPS coords in the KI (must also have their KI calibrated)
+   VAR BOOL     GPSEnabled[1]     DEFAULT=0 DEFAULTOPTION=VAULT
```
Comment (Hoikas) [inline Scripts/SDL/Personal.sdl:2486]: Whitespace around `=`

File: Scripts/Python/psnlYeeshaPageChanges.py [hunk 6]
```diff
@@ -218,17 +219,17 @@ def EnableDisable(self, val):
         if val in self.enabledStateList:
             if PageNumber.value == 10:
                 PtDebugPrint("psnlYeeshaPageChanges: Attempting to enable drawing and collision on %s..." % self.sceneobject.getName())
+            if PageNumber.value == 26:
+                if self.enabledStateList == [1, 3]:
+                    PtFogSetDefLinear(0,0,0)
+                    PtSetClearColor(0,0,0)
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageChanges.py:225]:
```suggestion
                    PtFogSetDefLinear(0, 0, 0)
                    PtSetClearColor(0, 0, 0)
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+hideObjectsFalse = ptAttribSceneobjectList(4, "Hide Objects in List when States are false")
+
+
+class psnlYeeshaPageVisController(ptMultiModifier):
+    def __init__(self):
+        ptMultiModifier.__init__(self)
+        self.id = 53000
+        version = 0
+        self.version = version
+        PtDebugPrint("__init__psnlYeeshaPageVisController v%d.%d" % (version,1),level=kWarningLevel)
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:62]:
```suggestion
        PtDebugPrint(f"__init__psnlYeeshaPageVisController v{version}", level=kWarningLevel)
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+ *==LICENSE==* """
+
+from Plasma import *
+from PlasmaTypes import *
+from PlasmaVaultConstants import *
+from PlasmaNetConstants import *
+from xPsnlVaultSDL import *
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:48]:
```suggestion
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+PageNumber = ptAttribInt(1, "Yeesha Page Number")
+stringShowStates = ptAttribString(2,"States in which shown")
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:51]:
```suggestion
stringShowStates = ptAttribString(2, "States in which shown")
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+#~ (YeeshaPage25) Bahro poles/totems
+#~ (YeeshaPage26) Veelay Tsahvahn night skybox
+
+#Meaning of SDL values for each Yeesha Page:
+#
+#~ 0 - Page not found
+#~ 1 - Page found and active
+#~ 2 - Page found and inactive
+#~ 3 - Page found, active, and pending inactive when age emptied
+#~ 4 - Page found, inactive, and pending active when age emptied
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:99]: Can we move this outside of the class definition? As it stands, this breaks the indentation rules and may suddenly become an error in the future.

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+
+    def OnFirstUpdate(self):
+            try:
+                self.enabledStateList = stringShowStates.value.split(",")
+                for i in range(len(self.enabledStateList)):
+                    self.enabledStateList[i] = int(self.enabledStateList[i].strip())
+            except:
+                PtDebugPrint("xAgeSDLIntActEnabler.OnFirstUpdate():\tERROR: couldn't process start state list")
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:108]: One block = 4 spaces

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+#~ 2 - Page found and inactive
+#~ 3 - Page found, active, and pending inactive when age emptied
+#~ 4 - Page found, inactive, and pending active when age emptied
+
+
+    def OnFirstUpdate(self):
+            try:
+                self.enabledStateList = stringShowStates.value.split(",")
+                for i in range(len(self.enabledStateList)):
+                    self.enabledStateList[i] = int(self.enabledStateList[i].strip())
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:106]:
```suggestion
                self.enabledStateList = [int(i.strip()) for i in stringShowStates.value.split(",")]
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+            except:
+                PtDebugPrint("xAgeSDLIntActEnabler.OnFirstUpdate():\tERROR: couldn't process start state list")
+
+
+    def OnServerInitComplete(self):
+        
+        AgeVault = ptAgeVault()
+        if AgeVault is not None: #is the Vault online?
+
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:115]:
```suggestion
        if AgeVault := ptAgeVault():
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+            self.ageSDL = AgeVault.getAgeSDL()
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:116]: Let's not cache this.

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+                    SDLVar = self.ageSDL.findVar("YeeshaPage" + str(PageNumber.value))
+                    CurrentValue = SDLVar.getInt()
+                    #PtDebugPrint("psnlYeeshaPageVisController.OnServerInitComplete:\tYeeshaPage%d = %d" % (PageNumber.value, SDLVar.getInt()))
+                except:
+                    PtDebugPrint("psnlYeeshaPageVisController:\tERROR reading age SDLVar. Assuming CurrentValue = 0")
+                    CurrentValue = 0
+
+                self.EnableDisable(CurrentValue)
+            else:
+                PtDebugPrint("psnlYeeshaPageVisController: Error trying to access the Chronicle self.ageSDL. self.ageSDL = %s" % ( self.ageSDL))
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:128]: Age SDL isn't a chronicle...???

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+        else:
+            PtDebugPrint("psnlYeeshaPageVisController: Error trying to access the Vault. Can't access YeeshaPageChanges chronicle.")
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:130]: Yeesha pages don't use chronicles...

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+
+
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:132]:
```suggestion
```

File: Scripts/Python/psnlYeeshaPageChanges.py [hunk 18]
```diff
@@ -269,18 +265,24 @@ def TimeToGrow(self):
...
+                    SDLVar.setInt( (size * 10) + state)
+                    AgeVault.updateAgeSDL(ageSDL)
+                elif sizechanged:
+                    SDLVar.setInt( (size * 10) + state)
+                    AgeVault.updateAgeSDL(ageSDL) 
+                return state
+            else:
+                PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the ageSDL. ageSDL = %s" % ( ageSDL))
+        else:
+            PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the Vault.")
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageChanges.py:288]: It looks like we already have the age vault, so we can simplify this by:

```suggestion
        if ageSDL := AgeVault.getAgeSDL():
            if state == 3:
                state = 2
                PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 3 to 2." % ("YeeshaPage" + str(PageNumber.value)))
                SDLVar.setInt( (size * 10) + state)
                AgeVault.updateAgeSDL(ageSDL) 
            elif state == 4 or state > 4:
                state = 1
                PtDebugPrint("psnlYeeshaPageChanges: Updated value of YeeshaPage %s from 4 to 1." % ("YeeshaPage" + str(PageNumber.value)))
                SDLVar.setInt( (size * 10) + state)
                AgeVault.updateAgeSDL(ageSDL)
            elif sizechanged:
                SDLVar.setInt( (size * 10) + state)
                AgeVault.updateAgeSDL(ageSDL) 
            return state
        else:
            PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the ageSDL. ageSDL = %s" % ( ageSDL))
    else:
        PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the Vault.")
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+        version = 0
+        self.version = version
+        PtDebugPrint(f"__init__psnlYeeshaPageVisController v{version}", level=kWarningLevel)
+
+    def OnFirstUpdate(self):
+        try:
+            self.enabledStateList = [int(i.strip()) for i in stringShowStates.value.split(",")]
+        except:
+            PtDebugPrint("xAgeSDLIntActEnabler.OnFirstUpdate():\tERROR: couldn't process start state list")
+
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:104]:
```suggestion
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+
+    def OnServerInitComplete(self):
+
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:107]:
```suggestion
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+
+    def OnServerInitComplete(self):
+        
+        if AgeVault := ptAgeVault():
+            if ageSDL := AgeVault.getAgeSDL():
+                try:
+                    SDLVar = ageSDL.findVar("YeeshaPage" + str(PageNumber.value))
+                    CurrentValue = SDLVar.getInt()
+                    #PtDebugPrint("psnlYeeshaPageVisController.OnServerInitComplete:\tYeeshaPage%d = %d" % (PageNumber.value, SDLVar.getInt()))
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:113]:
```suggestion
                    PtDebugPrint(f"psnlYeeshaPageVisController.OnServerInitComplete:\t{YeeshaPagePageNumber.value} = {SDLVar.getInt()}", level=kDebugDumpLevel)
```

**Comment by Hoikas** [comment]
@Hazado Could you rebase this on top of the latest master?

**Comment by Hazado** [comment]
> @Hazado Could you rebase this on top of the latest master?

Done

File: Scripts/Python/psnlYeeshaPageChanges.py [hunk 22]
```diff
@@ -63,19 +63,10 @@
 respDisable = ptAttribResponder(6, "Disabled resp (if necessary)")
 
 #globals
-TotalPossibleYeeshaPages = 25
+TotalPossibleYeeshaPages = 26
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageChanges.py:66]: Might be more future-proof to:
```suggestion
import xLinkingBookDefs
TotalPossibleYeeshaPages = len(xLinkingBookDefs.xYeeshaPages)
```

So this value doesn't have to be updated when new pages are added.

File: Scripts/Python/psnlYeeshaPageChanges.py [hunk 23]
```diff
@@ -113,13 +105,20 @@ def __init__(self):
...
-            try:
-                self.enabledStateList = stringShowStates.value.split(",")
-                for i in range(len(self.enabledStateList)):
-                    self.enabledStateList[i] = int(self.enabledStateList[i].strip())
-            except:
-                PtDebugPrint("xAgeSDLIntActEnabler.OnFirstUpdate():\tERROR: couldn't process start state list")
+        try:
+            self.enabledStateList = [int(i.strip()) for i in stringShowStates.value.split(",")]
+        except:
+            PtDebugPrint("xAgeSDLIntActEnabler.OnFirstUpdate():\tERROR: couldn't process start state list")
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageChanges.py:121]:
```suggestion
            PtDebugPrint("psnlYeeshaPageChanges.OnFirstUpdate():\tERROR: couldn't process start state list")
```

File: Scripts/Python/psnlYeeshaPageChanges.py [hunk 24]
```diff
@@ -203,32 +199,32 @@ def OnServerInitComplete(self):
                 if PageNumber.value == 5 and stringShowStates.value == "0":
                     PtDebugPrint("psnlYeeshaPageChanges: You've found the following Yeesha Pages:")
                     for thispage in range(1,TotalPossibleYeeshaPages+1):
-                        FoundValue = self.ageSDL.findVar("YeeshaPage" + str(thispage))
+                        FoundValue = ageSDL.findVar("YeeshaPage" + str(thispage))
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageChanges.py:202]:
```suggestion
                        FoundValue = ageSDL.findVar(f"YeeshaPage{thispage}")
```

File: Scripts/Python/psnlYeeshaPageChanges.py [hunk 25]
```diff
@@ -203,32 +199,32 @@ def OnServerInitComplete(self):
...
                         PtDebugPrint ("\t The previous value of the SDL variable %s is %s" % ("YeeshaPage" + str(thispage), FoundValue.getInt()))
                         if FoundValue is not None and FoundValue.getInt() != 0: 
                             PtDebugPrint ("psnlYeeshaPageChanges: You have found Yeesha Page # %s." % (thispage))
                             
             else:
-                PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the Chronicle self.ageSDL. self.ageSDL = %s" % ( self.ageSDL))
+                PtDebugPrint("psnlYeeshaPageChanges: Error trying to access the ageSDL. ageSDL = %s" % ( ageSDL))
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageChanges.py:208]:
```suggestion
                PtDebugPrint(f"psnlYeeshaPageChanges: Error trying to access the ageSDL. ageSDL = {ageSDL}")
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 26]
```diff
@@ -0,0 +1,130 @@
...
+        try:
+            self.enabledStateList = [int(i.strip()) for i in stringShowStates.value.split(",")]
+        except:
+            PtDebugPrint("xAgeSDLIntActEnabler.OnFirstUpdate():\tERROR: couldn't process start state list")
+
+    def OnServerInitComplete(self):
+        if AgeVault := ptAgeVault():
+            if ageSDL := AgeVault.getAgeSDL():
+                try:
+                    SDLVar = ageSDL.findVar("YeeshaPage" + str(PageNumber.value))
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:109]:
```suggestion
                    SDLVar = ageSDL.findVar(f"YeeshaPage{PageNumber.value}")
```

File: Scripts/Python/psnlYeeshaPageVisController.py [hunk 27]
```diff
@@ -0,0 +1,130 @@
...
+        self.id = 53000
+        version = 0
+        self.version = version
+        PtDebugPrint(f"__init__psnlYeeshaPageVisController v{version}", level=kWarningLevel)
+
+    def OnFirstUpdate(self):
+        try:
+            self.enabledStateList = [int(i.strip()) for i in stringShowStates.value.split(",")]
+        except:
+            PtDebugPrint("xAgeSDLIntActEnabler.OnFirstUpdate():\tERROR: couldn't process start state list")
```
Comment (Hoikas) [inline Scripts/Python/psnlYeeshaPageVisController.py:103]:
```suggestion
            PtDebugPrint("psnlYeeshaPageVisController.OnFirstUpdate():\tERROR: couldn't process start state list")
```

**Pull request merged**